policies/codingstyle.txt

```diff
@@ -289,12 +289,69 @@ options.
 
 In source files, separate functions with one blank line.
 
+                Chapter 6.1: Function Arguments
+
+Most API functions in OpenSSL revolve around a specific type (e.g. SSL_CTX,
+EVP_PKEY, X509 etc). When adding new API functions relevant to a given type, the
+following argument ordering should be used:
+
+    rettype TYPE_get_something(TYPE *,[args])
+    rettype TYPE_do_something(TYPE *,[args])
+    TYPE *TYPE_new([args])
+
+Where present the TYPE argument should always be first. In constructors or
+object factory type functions (such as a "fetch" function), the key argument
+that identifies the thing be constructed should be placed first (if such an
+argument exists). For example in a "fetch" function the name of the algorithm
+being fetched would be first.
+
+The remaining argument list should be put into order of importance. Deciding the
+relative importance of arguments is necessarily somewhat subjective, but the
+following guidelines should be considered:
+
+1) Any "context" arguments should be first. If multiple contexts are necessary
+   then they should be in order of most general to most specific. In some cases
+   (typically with an OPENSSL_CTX), a context may be NULL to indicate a default
```

Comment:
Should this (already) reflect https://github.com/openssl/openssl/pull/12621?

Reply:
Good point. Yes, it probably should.

Reply:
Fixed